core/access/mod.go

```diff
@@ -3,33 +3,51 @@ package access
 
 import (
 	"encoding"
-	"strings"
 
+	"go.dedis.ch/dela/core/store"
 	"go.dedis.ch/dela/serde"
 )
 
 // Identity is an abstraction to uniquely identify a signer.
 type Identity interface {
 	serde.Message
-	encoding.TextMarshaler
-}
 
-// Access is an abstraction to verify if an identity has access to a
-// specific rule.
-type Access interface {
-	serde.Message
+	encoding.TextMarshaler
 
-	Match(rule string, idents ...Identity) error
+	Equal(other interface{}) bool
 }
 
-// Factory is the factory interface to deserialize accesses.
-type Factory interface {
-	serde.Factory
-
-	AccessOf(serde.Context, []byte) (Access, error)
+// Credentials is an abstraction of an entity that allows one or several
+// identities to access a given scope.
+//
+// As an example, the identifier is the username of a username/password pair. It
+// defines the component to compare against. Then the password is the list of
+// identities, verified beforehands, that will match, or won't, match to the
+// identifier underlying permissions. The rule defines which scope should be
+// verified so that the permissions can hold multiple of thoses.
+//
+//   []byte{0xde, 0xad, 0xbe, 0xef} -> {
+//     "myContract:sayHello": ["Alice", "Bob"],
+//     "myContract:sayBye": ["Bob"],
+//   }
+//
+// The example above shows an access control for the contract "myContract" that
+// is allowing two commands "sayHello" and "sayBye". Alice and Bob can say
+// hello, but only Bob is allow to say bye. Alice can prove that she's allowed
+// by providing credentials with the identifier 0xdeadbeef.
+type Credentials interface {
+	GetID() []byte
+	GetRule() string
```

Comment:
from the example above I am missing what GetRules() corresponds to. Should this not return a `[]string` ?

Reply:
No because you match a single string, which can be whatever the credentials defines it should be. In the case of a contract, it's the contract and the action concatenated. But that could be a simple word for something else.

Reply:
And in the case of the example above ?

Reply:
Oh I see where the confusion is. The key on the left is the rule, not the array of identities on the right.

Reply:
So in the example, the credentials are define with `0xdeadbeef`+`myContract:sayHello` or `0xdeadbeef` + `myContract:sayBye`

Reply:
Ok, so based on this example there would be 2 credential, right? Then maybe it would make more sense if `Credentials` is in singular ?

Reply:
I had this thought too so probably, yeah.

Reply:
Ok cool. Then it would help if you can update the example to explain what GetRule corresponds to

Reply:
The comment is also updated to avoid the confusion.